### PR TITLE
Add core unit tests (#170)

### PR DESCRIPTION
**Pull request by DannyvdSluijs**
Hey!

Type: code quality

Link to issue: #170 

This pull request affects the following components: **(please check boxes)**
- [X] Core
- [ ] Analyzer
- [ ] Compiler
- [ ] Control Flow Graph
- [ ] Documentation

**In raising this pull request, I confirm the following (please check boxes):**
- [X] I have read and understood the [Contributing Guidelines](/.github/CONTRIBUTING.md)?
- [X] I have checked that another pull request for this purpose does not exist.
- [X] I wrote some tests for this PR.

Small description of change: Added core unit tests for IssuesCollector, ScopePointer and Application

Thanks

### CODE REVIEW

File: tests/PHPSA/IssuesCollectorTest.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * phpsa
+ *
+ * PHP Version 7
+ *
+ * @link      http://scm.monitorlinq.com/backend
+ * @copyright Copyright (c) 2016 Monitorlinq Limited
+ * @license   http://www.monitorlinq.com/license Proprietary License
```
Comment (ddmler) [inline tests/PHPSA/IssuesCollectorTest.php]: whats this?

File: tests/PHPSA/ApplicationTest.php [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+{
+    /**
+     * @covers \PHPSA\Application::__construct
+     */
+    public function testConstructor()
+    {
+        $application = new Application();
+
+        $this->assertInstanceOf('\PHPSA\Application', $application);
+        $this->assertInstanceOf('\Symfony\Component\Console\Application', $application);
```
Comment (K-Phoen) [inline tests/PHPSA/ApplicationTest.php]: I don't know what @ovr and @ddmler think about that, but I don't really see the value in this kind of assertions.
Tests should reflect the behavior of the system, how it reacts given some input data, … Just testing that a class can be instantiated does not seem that useful to me.

File: tests/PHPSA/IssuesCollectorTest.php (hunk 1 above)
Comment (DannyvdSluijs) [inline tests/PHPSA/IssuesCollectorTest.php]: My mistake, will remove it and add to the PR.

File: tests/PHPSA/ApplicationTest.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Tests\PHPSA;
+
+use PHPSA\Application;
+
+class ApplicationTest extends TestCase
+{
+    /**
+     * @covers \PHPSA\Application::__construct
```
Comment (K-Phoen) [inline tests/PHPSA/ApplicationTest.php]: I know what it does, I'm just not sure that we actually want to use this feature :)

File: tests/PHPSA/ApplicationTest.php (hunk 2 above)
Comment (ovr) [inline tests/PHPSA/ApplicationTest.php]: I dont known, for me it's not need
It's rly simple tests that cannot give nothing, about BC I dont thing that someone will extends from \PHPSA\Application, but maybe...

**Comment by DannyvdSluijs** [comment]
Hi Guys,

I'm having some time to continue this. Any comments on what is needed to accept the PR?

BR

**Comment by ddmler** [comment]
I think if you remove the method testConstructor in ScopePointerTest and the ApplicationTest file we can merge.
There seems to be no useful test for application so we leave that out.

**Comment by DannyvdSluijs** [comment]
@ddmler Please find the additional commits removing the constructor test and support the changes merged from upstream.

**Comment by ddmler** [comment]
Thanks @DannyvdSluijs :+1:

**Comment by ovr** [comment]
@ddmler 

image

image

**Comment by ovr** [comment]
[phpsa] Branch "master" was force-pushed to `d8102bc` by ovr. [Compare changes](https://github.com/ovr/phpsa/compare/35ec2696895b...d8102bcedefd)

**Comment by ovr** [comment]
branch, was interactive rebased and cherry-picked into master without `merge` commits

image

**Comment by DannyvdSluijs** [comment]
@ovr Thanks for taking the issue. Could you show the specific steps/commands used to achieve the rebase. Rebasing is still somewhat unclear to me. Next time it would like to spare you the trouble.

**Comment by ddmler** [comment]
@ovr sorry didn't see the merge commit
@DannyvdSluijs https://git-scm.com/book/en/v2/Git-Branching-Rebasing
or check the image from ovr .. there is a typo it's remote instead of remove

**Comment by ovr** [comment]
> @ovr sorry didn't see the merge commit
@ddmler not a problem :)

@DannyvdSluijs , @ddmler write you link or your can review how to rebase in our documentation inside project

Sorry, for late reply :(

**Pull request merged**